src/pods/BasicPodManager.ts

```diff
@@ -0,0 +1,62 @@
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import type { ResourceStore } from '../storage/ResourceStore';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { UnsupportedHttpError } from '../util/errors/UnsupportedHttpError';
+import { ensureTrailingSlash } from '../util/PathUtil';
+import type { DataGenerator } from './DataGenerator';
+import type { PodManager } from './PodManager';
+import type { User } from './User';
+
+/**
+ * Pod manager that uses a {@link DataGenerator} to create the default resources for a new pod.
+ */
+export class BasicPodManager implements PodManager {
+  private readonly base: string;
+  private readonly store: ResourceStore;
+  private readonly generator: DataGenerator;
+
+  public constructor(base: string, store: ResourceStore, generator: DataGenerator) {
+    this.base = base;
+    this.store = store;
+    this.generator = generator;
+  }
+
+  /**
+   * Creates a new pod using all resources created by the data generator.
+   * Pod identifiers are created by appending the username to the base URI.
+   * Will throw an error if the given identifier already has a resource.
+   */
+  public async createPod(user: User): Promise<ResourceIdentifier> {
+    const identifier = this.generateIdentifier(user.username);
+    try {
+      const result = await this.store.getRepresentation(identifier, {});
+      result.data.destroy();
+      throw new ConflictHttpError(`There already is a resource at ${identifier.path}`);
+    } catch (error: unknown) {
+      // We want the identifier to not exist
+      if (!(error instanceof NotFoundHttpError)) {
+        throw error;
+      }
+    }
+
+    const data = await this.generator.generate(identifier, user);
+    // Sort to make sure containers are generated before documents
+    const paths = Object.keys(data).sort((left, right): number => left.length - right.length);
+    for (const path of paths) {
+      await this.store.setRepresentation({ path }, data[path]);
```

Comment:
Will be easier once we add a `deletePod` function, that could then be called. I initially had one but removed it again because I wasn't sure we really wanted such a function immediately (accidentally adding data is way less worse than accidentally deleting it).